src/com/projectkorra/projectkorra/airbending/flight/FlightMultiAbility.java

```diff
@@ -215,8 +243,8 @@ public void run() {
 			player.setFlying(true);
 		} else if (mode == FlightMode.ENDING) {
 			player.setGliding(hadGlide);
-			player.setAllowFlight(canFly);
-			player.setFlying(hadFly);
+			player.setAllowFlight(flight.getCouldFly());
```

Comment:
Again, FlightHandler will manage the reversion of a player's flight state. These lines need to be removed.

Reply:
Refer to first comment